src/Shared/Recruit.Vacancies.Client/Infrastructure/Reports/QaApplicationsReportStrategy.cs

```diff
@@ -0,0 +1,224 @@
+        }
+
+
+        private readonly IApprenticeshipProgrammeProvider _programmeProvider;
+        private readonly ITimeProvider _timeProvider;
+        private readonly ILogger<QaApplicationsReportStrategy> _logger;
+        private readonly IMongoCollection<BsonDocument> _collection;
+
+        private const string QueryFormat = @"[
+            { $match: { createdDate: { $gt: ISODate('_fromDate_'), $lte: ISODate('_toDate_')}} },
+            { $project: {
+                _id: 0
+                ,'Vacancy title': { $ifNull: ['$title', null] }
+                ,'Vacancy reference number': { $ifNull: ['$vacancyReference', null] }
+                ,'Submission number' : { $ifNull: ['$submissionCount', null] }
+                ,'Date submitted': { $ifNull: ['$createdDate', null] }
+                ,'SLA deadline': { $ifNull: ['$slaDeadline', null] }
+                ,'Review started': { $ifNull: ['$reviewedDate', null] }
+                ,'Review completed': { $ifNull: ['$closedDate', null] }
+                ,'Outcome': {$ifNull : ['$manualOutcome', null] }
+                ,'Number of issues reported': {
+                    $size: {
+                        $filter: {
+                            input: {$ifNull: ['$manualQaFieldIndicators', []] },
+                            as: 'field',
+                            cond: { $eq: ['$$field.isChangeRequested', true] }
+                        }
+                    }
+                }
+                ,'Reviewed by': { $ifNull: ['$reviewedByUser.name', null] }
+                ,'Vacancy submitted by': { $ifNull: ['$vacancySnapshot.ownerType', null] }
+                ,'Employer': { $ifNull: ['$vacancySnapshot.legalEntityName', null] }
+                ,'Display name': { $ifNull: ['$vacancySnapshot.employerName', null] }
+                ,'Training provider': { $ifNull: ['$vacancySnapshot.trainingProvider.name', null] }
+                ,'Vacancy postcode': { $ifNull: ['$vacancySnapshot.employerLocation.postcode', null] }
+                ,'programmeId': { $ifNull: ['$vacancySnapshot.programmeId', null] }
+                ,'Closure method': { $ifNull: ['', null] }
+                ,'Closed by': { $ifNull: ['', null]}
+            }}
+        ]";
+
+        public QaApplicationsReportStrategy(
+            ILoggerFactory loggerFactory,
+            IOptions<MongoDbConnectionDetails> details,
+            IApprenticeshipProgrammeProvider programmeProvider,
+            ITimeProvider timeProvider,
+            ILogger<QaApplicationsReportStrategy> logger)
+            : base(loggerFactory, MongoDbNames.RecruitDb, MongoDbCollectionNames.VacancyReviews, details)
+        {
+            _programmeProvider = programmeProvider;
+            _timeProvider = timeProvider;
+            _logger = logger;
+            _collection = GetCollection<BsonDocument>();
+            RetryPolicy = MongoDbRetryPolicy.GetConnectionRetryPolicy(_logger);
```

Comment:
Not sure this is right. What happens if a `MongoException` is thrown? (See other comment in Mongo DB Retry comment)